src/layers/core/polygon-layer/polygon-layer.js

```diff
@@ -73,14 +73,14 @@ export default class PolygonLayer extends CompositeLayer {
     }
   }
 
-  _onHoverSubLayer(info) {
-    info.object = (info.object && info.object.feature) || info.object;
-    this.props.onHover(info);
-  }
+  _onPickSubLayer(handler, info) {
+    Object.assign(info, {
+      layer: this,
+      // override object with picked feature
+      object: (info.object && info.object.feature) || info.object
```

Comment:
A way to make this work is if the sublayers keep a reference to the parent layer. Then in picking we can directly call the `getPickingInfo()` and `onHover` of the parent layer.